src/Spatial/Euclidean/Line3D.cs

```diff
@@ -181,6 +182,91 @@ public bool IsParallelTo(Line3D other, Angle angleTolerance)
         }
 
         /// <summary>
+        /// Computes the pair of points which represent the closest distance between this Line3D and another Line3D, with the first
+        /// point being the point on this Line3D, and the second point being the corresponding point on the other Line3D.  If the lines
+        /// intersect the points will be identical, if the lines are parallel the first point will be the start point of this line.
+        /// </summary>
+        /// <param name="other">line to compute the closest points with</param>
+        /// <returns>A tuple of two points representing the endpoints of the shortest distance between the two lines</returns>
+        public Tuple<Point3D, Point3D> ClosestPointsBetween(Line3D other)
```

Comment:
To me the chosen return type and naming does make sense. This is really about two points (which can be the same), not technically a line.

There could be another routine on top which does returns a line, but this other routine would then have to fail in some cases (e.g. when they intersect).